### PR TITLE
Add hardware report job

### PR DESCRIPTION
**Pull request by comloo**

### CODE REVIEW

File: dags/longitudinal.py [hunk 1]
```diff
@@ -40,11 +40,11 @@
                       job_name="Firefox Hardware Report",
                       execution_timeout=timedelta(hours=10),
```
Comment (fbertsch) [inline dags/longitudinal.py]: Can you confirm that this timeout is still appropriate?

File: jobs/hardware_report.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    exit 1
+fi
+
+# Create the package for distribution across the cluster
+git clone https://github.com/mozilla/python_mozetl.git
+cd python_mozetl
+python setup.py bdist_egg
+
+# Generate the driver script
+echo 'from mozetl.hardware_report import hardware_dashboard; hardware_dashboard.main()' > run.py
```
Comment (fbertsch) [inline jobs/hardware_report.sh]: This is different than the way we run the rest of the jobs. Is this the only way to run the hardware report? Doesn't this mean the input variables are ignored? (`bucket` and `date`).

File: dags/longitudinal.py (hunk 1 above)
Comment (comloo) [inline dags/longitudinal.py]: The job takes about 3-6 hours to run, so I think 10 hour timeout is appropriate.

File: jobs/hardware_report.sh (hunk 2 above)
Comment (whd) [inline jobs/hardware_report.sh]: This appears to be using a new(ish) idiom for python ETL jobs that is also used by churn, engagement ratio, search, and the topline reports.  That said, we're indeed missing an `env` section in the operator definition to propagate the environment variables as arguments.

File: dags/longitudinal.py [hunk 3]
```diff
@@ -40,11 +40,11 @@
...
                       instance_count=15,
```
Comment (whd) [inline dags/longitudinal.py]: Add a line like `env={"date": DS_WEEKLY, "bucket": "{{ task.__class__.public_output_bucket }}"},` here (see below).

**Pull request merged**